x-pack/plugins/enterprise_search/public/applications/app_search/components/crawler/components/crawler_status_indicator/start_crawl_context_menu.tsx

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import React, { useState } from 'react';
+
+import { useActions } from 'kea';
+
+import { EuiButton, EuiContextMenuItem, EuiContextMenuPanel, EuiPopover } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+
+import { CrawlerLogic } from '../../crawler_logic';
+
+import { CrawlSelectDomainsModalLogic } from '../crawl_select_domains_modal/crawl_select_domains_modal_logic';
+
+interface Props {
+  menuButtonLabel?: string;
+  fill?: boolean;
+}
+
+export const StartCrawlContextMenu: React.FC<Props> = ({ menuButtonLabel, fill }) => {
+  const { startCrawl } = useActions(CrawlerLogic);
+  const { showModal: showCrawlSelectDomainsModal } = useActions(CrawlSelectDomainsModalLogic);
+
+  const [isPopoverOpen, setPopover] = useState(false);
```

Comment:
question: Ooh, local state. Do we have any general standards on when to use local state vs kea/Redux?

Reply:
I've gotten feedback in the past from other devs here to use the [useState hook for cases like this where the component is just tracking piece of a single prop in its state, often a boolean like this tracking if a flyout/modal/popover/etc is open or closed.  It hasn't always sat right with me though, what do you think?

Reply:
I have no strong opinion here, either approach will work. There's something to be said for not overcomplicating your logic file with this kind of really local stuff, but once you need to hook into that stuff for your logic you'll likely want to move it. Was just looking for whether folks had landed on a standard here yet :)

Reply:
Yeah you can see for instance I use a boolean to track which the modal is open in a logic file.  I go back and forth whether boilerplate of logic file + tests is better than this 🙃